dataactvalidator/validation_handlers/validationManager.py

```diff
@@ -416,17 +430,73 @@ def load_file_data(self, sess, bucket_name, region_name):
 
         return file_row_count
 
-    def process_data_chunk(self, sess, chunk_df):
+    def parallel_data_loading(self, reader_obj):
+        with Manager() as server_manager:
+            # These variables will need to be shared among the processes and used later overall
+            shared_data = server_manager.dict()
+            shared_data['total_rows'] = self.total_rows
+            shared_data['has_data'] = self.has_data
+            shared_data['error_rows'] = self.error_rows
+            shared_data['error_list'] = self.error_list
+            shared_data['header_dict'] = self.reader.header_dict
+            shared_data['flex_fields'] = self.reader.flex_fields
+            # setting reader to none as multiprocess can't pickle it, it'll get reset
+            temp_reader = self.reader
+            self.reader = None
+
+            m_lock = server_manager.Lock()
+            pool = Pool(MULTIPROCESSING_POOLS)
+            for chunk_df in reader_obj:
+                pool.apply_async(func=self.process_data_chunk, args=(chunk_df, shared_data, m_lock))
+            pool.close()
+            pool.join()
+
+            # Resetting these out here as they are used later in the process
+            self.total_rows = shared_data['total_rows']
+            self.has_data = shared_data['has_data']
+            self.error_rows = shared_data['error_rows']
+            self.error_list = shared_data['error_list']
+            self.reader = temp_reader
+
+    def iterative_data_loading(self, reader_obj):
+        shared_data = {
+            'total_rows': self.total_rows,
+            'has_data': self.has_data,
+            'error_rows': self.error_rows,
+            'error_list': self.error_list,
+            'header_dict': self.reader.header_dict,
+            'flex_fields': self.reader.flex_fields
+        }
+        for chunk_df in reader_obj:
+            self.process_data_chunk(chunk_df, shared_data)
+
+        # Resetting these out here as they are used later in the process
+        self.total_rows = shared_data['total_rows']
+        self.has_data = shared_data['has_data']
+        self.error_rows = shared_data['error_rows']
+        self.error_list = shared_data['error_list']
+
+    def process_data_chunk(self, chunk_df, shared_data, m_lock=None):
         """ Loads in a chunk of the file and performs initial validations
 
             Args:
-                sess: the database connection
                 chunk_df: the chunk of the file to process as a dataframe
+                shared_data: dictionary of shared data among the chunks
+                m_lock: manager lock if provided to ensure processes don't override each other
         """
+        if m_lock:
+            # make a new connection per process
+            conn = db_connection()
+            sess = conn.session
+            lockable = m_lock
+        else:
+            sess = GlobalDB.db().session
+            lockable = NoLock()
```

Comment:
good to put a comment here that says you're adhering to the locking pattern with a no-op object that bypasses any actual locking